scipy/_lib/_array_api.py

```diff
@@ -652,3 +657,115 @@ def xp_default_dtype(xp):
 def is_marray(xp):
     """Returns True if `xp` is an MArray namespace; False otherwise."""
     return "marray" in xp.__name__
+
+
+
+def _make_capabilities(skip_backends=None, cpu_only=False):
+    skip_backends = [] if skip_backends is None else skip_backends
+
+    capabilities = dataclasses.make_dataclass('capabilities', ['cpu', 'gpu', 'mask'])
+
+    numpy = capabilities(cpu=True, gpu=False, mask=True)
+    strict = capabilities(cpu=True, gpu=False, mask=False)
+    cupy = capabilities(cpu=False, gpu=True, mask=False)
+    torch = capabilities(cpu=True, gpu=True, mask=False)
+    jax = capabilities(cpu=True, gpu=True, mask=False)
+    dask = capabilities(cpu=True, gpu=True, mask=False)
+
+    capabilities = dict(numpy=numpy, array_api_strict=strict, cupy=cupy,
+                        torch=torch, jax=jax, dask=dask)
+
+    for backend, _ in skip_backends:  # ignoring the reason
+        setattr(capabilities[backend], 'cpu', False)
+        setattr(capabilities[backend], 'gpu', False)
+        setattr(capabilities[backend], 'mask', False)
+
+    if cpu_only:
+        cupy.gpu = False
+        torch.gpu = False
+        jax.gpu = False
+        dask.gpu = False
+
+    return capabilities
+
+
+def _make_capabilities_table(capabilities):
+    numpy = capabilities['numpy']
+    cupy = capabilities['cupy']
+    torch = capabilities['torch']
+    jax = capabilities['jax']
+    dask = capabilities['dask']
+
+    for backend in [numpy, cupy, torch, jax, dask]:
+        for attr in ['cpu', 'gpu', 'mask']:
+            val = "✓" if getattr(backend, attr) else " "
+            val += " " * (len('{numpy.}') + len(attr) - 1)
+            setattr(backend, attr, val)
+
+    table = f"""                
+    +---------+-------------+-------------+--------------+
+    | Library | CPU         | GPU         | ``marray``   |
```

Comment:
Certainly not required for this pr but it could be nice in the future if the `marray` text and other libraries linked to their docs

Reply:
Yeah I thought about that, but I'm not sure reST tables support links. I tried citations before, and that didn't work.